internal/healthcheck/healthcheck.go

```diff
@@ -88,8 +89,24 @@ func (c *Check) HandleReadiness(w http.ResponseWriter, _ *http.Request) {
 		return
 	}
 
-	err := c.proxy.CheckConnections(ctx)
-	if err != nil {
+	var minReady int
+	q := req.URL.Query()
+	if v := q.Get("min-ready"); v != "" {
+		n, err := strconv.Atoi(v)
+		// If the query value parsed correctly, set it to minReady. Otherwise,
+		// ignore it.
+		if err == nil {
+			minReady = n
+		}
+	}
+
+	n, err := c.proxy.CheckConnections(ctx)
+	if minReady > n {
+		c.logger.Errorf("[Health Check] min-ready was greater than registered instances")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if err != nil && !ready(err, minReady, n) {
```

Comment:
This is kind of a strange condition statement. It took me a few minutes to figure out why it's ok to return an error if err != nil and minReady == 0. 

I'd move the `err != nil` condition inside ready() so that the logic reads more like a story and less like a CS 101 puzzle.

Reply:
That seems reasonable. I'll update this.